actix-web-grants/src/guards.rs

```diff
@@ -29,22 +29,57 @@ use std::hash::Hash;
 ///    Ok(HashSet::from(["ROLE_ADMIN".to_string()]))
 /// }
 /// ```
-pub struct AuthorityGuard<Type> {
-    allow_authority: Type,
+
+pub struct AuthorityGuard<T> {
+    allow_authority: Type<T>,
+}
+
+pub enum Type<T> {
+    Single(T),
+    Any(Vec<T>),
+    All(Vec<T>),
 }
 
-impl<Type: Eq + Hash + 'static> AuthorityGuard<Type> {
-    pub fn new(allow_authority: Type) -> AuthorityGuard<Type> {
-        AuthorityGuard { allow_authority }
+impl<T: Eq + Hash + 'static> AuthorityGuard<T> {
+     pub fn create(allow_authority: Type<T>) -> AuthorityGuard<T> {
+        AuthorityGuard {
+            allow_authority: allow_authority,
+        }
+    }
```

Comment:
```suggestion
     fn create(allow_authority: Type<T>) -> AuthorityGuard<T> {
        AuthorityGuard {
            allow_authority: allow_authority,
        }
    }
```

I guess there is no need to expose this one